src/ai/backend/agent/agent.py

```diff
@@ -2369,12 +2369,19 @@ async def create_kernel(
                     raise AgentError(
                         f"Cancelled waiting of container startup (k:{str(ctx.kernel_id)}, container:{container_data['container_id']})"
                     )
-                except Exception:
+                except BaseException as e:
                     log.exception(
-                        "unexpected error while waiting container startup (k:{})", kernel_id
+                        f"unexpected error while waiting container startup (k: {kernel_id}, e: {repr(e)})"
                     )
-                    raise RuntimeError(
-                        "cancelled waiting of container startup due to initialization failure",
+                    await self.inject_container_lifecycle_event(
+                        kernel_id,
+                        session_id,
+                        LifecycleEvent.DESTROY,
+                        KernelLifecycleEventReason.FAILED_TO_START,
+                        container_id=ContainerId(container_data["container_id"]),
+                    )
+                    raise AgentError(
+                        f"Container startup failed (k:{str(ctx.kernel_id)}, container:{container_data['container_id']}, e:{repr(e)})"
                     )
```

Comment:
The change from BaseException to AgentError is a bit awkward.

Reply:
Managers can handle `AgentError`s and change the status of session to canceled

Reply:
My fault, any exception is handled by managers. I updated this part to re-raise the exception